File: current_not_used/stream/app/src/main/java/com/makeus/kkongi/streampractice/MainActivity.java
package com.makeus.kkongi.streampractice;

import android.os.Bundle;
import android.os.PersistableBundle;
import android.util.Log;

import androidx.annotation.Nullable;
import androidx.appcompat.app.AppCompatActivity;

import java.util.Arrays;
import java.util.List;
import java.util.stream.Collectors;

public class MainActivity extends AppCompatActivity {
    @Override
    public void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        List<Integer> numbers1 = Arrays.asList(1, 2, 3);
        List<Integer> numbers2 = Arrays.asList(3, 4);
        List<int[]> pairs = numbers1.stream()
                .flatMap(n1 ->
                        numbers2.stream().filter(n2 -> (n1 + n2) % 3 == 0).map(n2 -> new int[]{n1, n2}))
                .collect(Collectors.toList());
    }
}

File: app/src/main/java/com/jo/annotationprocessorpractice/MainActivity.kt
package com.jo.annotationprocessorpractice

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import com.jo.annotation.CharlesIntent

@CharlesIntent
class MainActivity : AppCompatActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        startActivity(Charles.intentForMainActivity(this))
    }
}